core/resources/polygon.vs

```diff
@@ -1,30 +1,30 @@
-#ifdef GL_ES
-precision mediump float;
-#endif
-
+uniform mat4 u_modelView;
 uniform mat4 u_modelViewProj;
-uniform vec3 u_lightDirection;
+uniform float u_time;
+
+#pragma tangram: material
+#pragma tangram: vert_lighting
 
 attribute vec4 a_position;
 attribute vec4 a_color;
 attribute vec3 a_normal;
 attribute vec2 a_texcoord;
 
-varying vec4 v_pos;
 varying vec4 v_color;
+varying vec3 v_eyeToPoint;
 varying vec3 v_normal;
 varying vec2 v_texcoord;
 
 void main() {
 
-	v_normal = a_normal;
+  	v_normal = normalize(a_normal);
+  	v_texcoord = a_texcoord;
 
-  	v_color = a_color;
-  	
-  	float lit = dot(normalize(u_lightDirection), normalize(a_normal));
-  	v_color.rgb *= clamp(lit * 1.5, 0.5, 1.5);
-  	
-	v_texcoord = a_texcoord;
+	v_eyeToPoint = vec3(u_modelView * a_position);
+	calculateLighting(v_eyeToPoint,v_normal);
```

Comment:
is not the same code neither function between vert/frag. Data is passed by

varying vec4 v_amb;
varying vec4 v_diff;
varying vec4 v_spec;

At the begining calculateLighting return a computed color... but that over-ride the original color and what ever get shadowed on vertex will remain shadowed for ever (very poetic right?)